dali/pipeline/operators/arithmetic/expression_impl_factory_cpu.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "dali/pipeline/operators/arithmetic/expression_tree.h"
+#include "dali/pipeline/operators/arithmetic/arithmetic_meta.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/operators/arithmetic/expression_impl_cpu.h"
+#include "dali/pipeline/operators/arithmetic/expression_impl_factory.h"
+
+#define ALLOWED_TYPES \
+  (uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float16, float, double)
+
+#define ALLOWED_OPS \
+  (ArithmeticOp::add, ArithmeticOp::sub, ArithmeticOp::mul, ArithmeticOp::div, ArithmeticOp::mod)
+
+namespace dali {
+
+std::unique_ptr<ExpressionImplBase> ExprImplFactory(const HostWorkspace &ws,
+                                                    const ExprNode &expr) {
+  std::unique_ptr<ExpressionImplBase> result;
+  DALI_ENFORCE(expr.GetNodeType() == NodeType::Function, "Only function nodes can be executed.");
+
+  switch (expr.GetSubexpressionCount()) {
+    case 2: {
+      auto op = NameToOp(expr.GetOp());
```

Comment:
this case seems big enough to put it in an auxiliary function. Any obstacles to doing that?